modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/index/contributor/ObjectEntryVersionModelIndexerWriterContributor.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.internal.search.spi.model.index.contributor;
+
+import com.liferay.object.model.ObjectEntryVersion;
+import com.liferay.object.service.ObjectEntryVersionLocalService;
+import com.liferay.portal.search.batch.BatchIndexingActionable;
+import com.liferay.portal.search.batch.DynamicQueryBatchIndexingActionableFactory;
+import com.liferay.portal.search.spi.model.index.contributor.ModelIndexerWriterContributor;
+import com.liferay.portal.search.spi.model.index.contributor.helper.ModelIndexerWriterDocumentHelper;
+
+/**
+ * @author Mikel Lorza
+ */
+public class ObjectEntryVersionModelIndexerWriterContributor
+	implements ModelIndexerWriterContributor<ObjectEntryVersion> {
+
+	public ObjectEntryVersionModelIndexerWriterContributor(
+		DynamicQueryBatchIndexingActionableFactory
+			dynamicQueryBatchIndexingActionableFactory,
+		ObjectEntryVersionLocalService objectEntryVersionLocalService) {
+
+		_dynamicQueryBatchIndexingActionableFactory =
+			dynamicQueryBatchIndexingActionableFactory;
+		_objectEntryVersionLocalService = objectEntryVersionLocalService;
+	}
+
+	@Override
+	public void customize(
+		BatchIndexingActionable batchIndexingActionable,
+		ModelIndexerWriterDocumentHelper modelIndexerWriterDocumentHelper) {
+
```

Comment:
You'll probably need to implement some check for the `enableIndexSearch` property here. Something like [this](https://github.com/liferay/liferay-portal/blob/46f2383c501a744fefc241cee2cb5ea01c901ec8/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L5268).